paimon-core/src/main/java/org/apache/paimon/operation/FileStoreExpireImpl.java

```diff
@@ -206,13 +229,16 @@ private void expireUntil(long earliestId, long endExclusiveId) {
                         snapshotManager
                                 .snapshot(endExclusiveId)
                                 .dataManifests(snapshotDeletion.manifestList()));
+        skipDeletion.addAll(
+                collectManifestFilesFromTags(beginInclusiveId, endExclusiveId, taggedSnapshots));
```

Comment:
`List<String> collectManifestSkippingSet`, it can contain manifest file name and manifest list name.